cppcache/src/ClientMetadataService.cpp

```diff
@@ -143,7 +144,13 @@ void ClientMetadataService::getClientPRMetadata(const char* regionFullPath) {
                                               reply.getFpaSet());
       if (m_bucketWaitTimeout > std::chrono::milliseconds::zero() &&
           reply.getNumBuckets() > 0) {
-        WriteGuard guard(m_PRbucketStatusLock);
+        //ReadGuard guard( m_PRbucketStatusLock );
+        std::unique_lock<std::timed_mutex> guard(m_timedBucketStatusLock, std::chrono::milliseconds(1000));
```

Comment:
What is the rational for hard coded 1s (1000ms) timeout?

Consider using the value of chrono and not doing math yourself.
```std::chrono::seconds(1)```

Formatting rules should be obeyed. Use clang-format, a .clang-format file exists to apply style rules.